homeassistant/components/epson/config_flow.py

```diff
@@ -0,0 +1,54 @@
+"""Config flow for epson integration."""
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_PORT
+
+from . import validate_projector
+from .const import DOMAIN
+from .exceptions import CannotConnect
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        vol.Required(CONF_NAME, default=DOMAIN): str,
+        vol.Required(CONF_PORT, default=80): int,
+    }
+)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for epson."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    async def async_step_import(self, import_config):
+        """Import a config entry from configuration.yaml."""
+        return await self.async_step_user(import_config)
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        errors = {}
+        if user_input is not None:
+            if self.host_already_configured(user_input[CONF_HOST]):
```

Comment:
You can drop this check once you set a unique ID. Home Assistant will automatically replace the old one if it finds a duplicate unique ID.

Reply:
I'm comparing if same host/ip already exists, not unique id. I don't think Home Assistant checking for duplicate configuration, right? Without it I can add same host multiple times.

Reply:
We do that check based on the unique ID. IPs can change since it's based on DHCP leases.

Reply:
so there are 3 options:
- don't care about possible duplicates; at least for now, I can start working after this PR on adding TCP connection, which would allow to add extra sensors and some unique_id of the projector
- use CONF_HOST as unique_id (IP can change, but default hostname of the projector is it's first half of MAC address, so it's unique)
- use same method as Denon AVR integration to get_mac.

For now I'm going with first option, I can change according to your opinion.